Remove ANSI colors when writing to external warnings file. (#11624)
sphinx-doc/sphinx: CHANGES.rst

  Patch by Vinay Sajip.
* #11622: Ensure that the order of keys in ``searchindex.js`` is deterministic.
  Patch by Pietro Albini.

> * #11617: ANSI control sequences are stripped from the output when writing to
>   an external file.

Reviewer: ```suggestion
* #11617: ANSI control sequences are stripped from the output when writing to
  a warnings file with :option:`-w <sphinx-build -w>`.
```
